entproto/service.go

```diff
@@ -39,19 +39,43 @@ var (
 	errNoServiceDef = errors.New("entproto: annotation entproto.Service missing")
 )
 
+type ServiceOptionsEnum int
+
+const (
+	// OptionGenGetAll generates a GetAll RPC for retrieving all ids of existing entities.
+	OptionGenGetAll ServiceOptionsEnum = iota
+)
+
+type serviceOptions struct {
+	GenerateGetAll bool
+}
+
 type service struct {
 	Generate bool
+	Options  *serviceOptions
 }
 
 func (service) Name() string {
 	return ServiceAnnotation
 }
 
-func Service() schema.Annotation {
-	return service{Generate: true}
+func Service(options ...ServiceOptionsEnum) schema.Annotation {
+	newService := &service{
+		Generate: true,
+		Options:  &serviceOptions{},
+	}
+
+	for _, option := range options {
+		switch option {
```

Comment:
Linter is failing me for this switch case because it only has one case. The idea behind the switch case is that the options are extendable beyond this.